src/dmd/libomf.d

```diff
@@ -167,33 +167,38 @@ final class LibOMF : Library
 
     /*****************************************************************************/
 
-    void addSymbol(OmfObjModule* om, const(char)* name, int pickAny = 0)
+    void addSymbol(OmfObjModule* om, const(char)[] name, int pickAny = 0)
     {
+        assert(name.length == strlen(name.ptr));
```

Comment:
@WalterBright : Might as well keep the pointer if we have to do this kind of checks...

Reply:
My assumption was that it was a stopgap solution, and that it could be removed once arrayification was completed. A good chunk of this PR could be removed if `ObjOmfSymbol` had a `char[]` instead of a `char*`.

Reply:
The strings stored in the library are 0 terminated.